app/models/token.rb

```diff
@@ -25,18 +25,22 @@ def last_used_date
   def decrease_tokens
     if self.tokens > 0
       self.tokens = self.tokens - 1
-      self.last_token_used_date = Date.today
+      self.last_token_used_date = Time.now
     end
     self.save
   end
 
-  def reassign_tokens_if_new_day
-    if self.last_token_used_date and self.last_token_used_date < Date.today
+  def reassign_tokens_if_after_regen_period(regeneration_period)
+    if self.last_token_used_date
+      if (self.last_token_used_date.to_i + regeneration_period*60*60) <= Time.now.to_i
```

Comment:
This logic seems a bit off. First, it's the "token regenerates X hours after using it" rather than "all tokens regenerate every X hours, offset from a fixed time." I don't mind that *so* much, except it seems from above that each time a test is run, the "last_token_used_date" attribute is updated. This seems to mean that the tokens won't regenerate until "X hours after the *last* token is used", which I definitely don't want.